module/epochs/qc_client.go

```diff
@@ -31,37 +33,32 @@ const (
 // QCContractClient is a client to the Quorum Certificate contract. Allows the client to
 // functionality to submit a vote and check if collection node has voted already.
 type QCContractClient struct {
-	nodeID            flow.Identifier  // flow identifier of the collection node
-	qcContractAddress string           // QuorumCertificate contract address
-	accountKeyIndex   uint             // account key index
-	signer            sdkcrypto.Signer // signer used to sign vote transaction
+	BaseContractClient
+
+	nodeID flow.Identifier // flow identifier of the collection node
 
-	account *sdk.Account // account belonging to the collection node
-	client  module.SDKClientWrapper
+	env templates.Environment
 }
 
 // NewQCContractClient returns a new client to the Quorum Certificate contract
-func NewQCContractClient(flowClient module.SDKClientWrapper, nodeID flow.Identifier, accountAddress string,
-	accountKeyIndex uint, qcContractAddress string, signer sdkcrypto.Signer) (*QCContractClient, error) {
-
-	// get account for given address
-	account, err := flowClient.GetAccount(context.Background(), sdk.HexToAddress(accountAddress))
+func NewQCContractClient(log zerolog.Logger,
+	flowClient module.SDKClientWrapper,
+	nodeID flow.Identifier,
+	accountAddress string,
+	accountKeyIndex uint,
+	qcContractAddress string,
+	signer sdkcrypto.Signer) (*QCContractClient, error) {
+
+	base, err := NewBaseContractClient(log, flowClient, accountAddress, accountKeyIndex, signer, qcContractAddress)
```

Comment:
It's a good practice to namespace loggers by component so it's easier to search for them later. 

```suggestion
	base, err := NewBaseContractClient(log.With().Str("component", "qc_contract_client").Logger(), flowClient, accountAddress, accountKeyIndex, signer, qcContractAddress)
```

Reply:
Let's add this for the `dkg.Client` constructor as well